### PR TITLE
Make attachment captions markdown

### PR DESCRIPTION
**Pull request by hyperobject**
Expands on #22220 by adding `decoratedText` and mentions fields to attachment messages and rendering them in chat. This also required changing what parts of attachment messages are clickable in order to make mentions behave as expected.

### CODE REVIEW

File: shared/chat/conversation/attachment-fullscreen/index.desktop.tsx [hunk 1]
```diff
@@ -69,9 +69,9 @@ class _Fullscreen extends React.Component<Props & Kb.OverlayParentProps, State>
         <Kb.Box style={styles.container}>
           <Kb.HotKey hotKeys={this.hotKeys} onHotKey={this.onHotKey} />
           <Kb.Box style={styles.headerFooter}>
-            <Kb.Text lineClamp={2} type="BodySemibold" style={{color: Styles.globalColors.black, flex: 1}}>
+            <Kb.Markdown lineClamp={2} style={{flex: 1}} meta={{message: this.props.message}}>
```
Comment (songgao) [inline shared/chat/conversation/attachment-fullscreen/index.desktop.tsx]: Not a big deal but there's also `Styles.globalStyles.flexOne`

File: shared/constants/chat2/message.tsx [hunk 2]
```diff
@@ -918,6 +922,7 @@ const validUIMessagetoMessage = (
     }
     case RPCChatTypes.MessageType.attachmentuploaded: // fallthrough
     case RPCChatTypes.MessageType.attachment: {
+      console.warn(m)
```
Comment (songgao) [inline shared/constants/chat2/message.tsx]: Did you mean to remove this?

Comment (hyperobject) [reply shared/constants/chat2/message.tsx]: whoops! good catch

File: shared/chat/conversation/attachment-fullscreen/index.desktop.tsx (hunk 1 above)
Comment (hyperobject) [reply shared/chat/conversation/attachment-fullscreen/index.desktop.tsx]: oh neat! i can change that, seems better for stylistic reasons

**Pull request merged**